scheduler/filter/port.go

```diff
@@ -35,7 +35,7 @@ func (p *PortFilter) portAlreadyInUse(node *cluster.Node, requested dockerclient
 	for _, c := range node.Containers() {
 		for _, port := range c.Info.HostConfig.PortBindings {
 			for _, binding := range port {
-				if binding.HostPort == requested.HostPort {
+				if binding.HostPort == requested.HostPort && binding.HostPort != "" && requested.HostPort != "" {
```

Comment:
nit:

Since we already asserted that `binding.HostPort` and `requested.HostPort` are the same, there is probably no need to check if both are `!= ""`.

Maybe only check `requested.HostPort`?

Otherwise, LGTM